datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -437,14 +436,22 @@ def activate_a_node(name: str, *, session: Session = Depends(get_session)):
                 ):  # pragma: no cover
                     downstream.current.status = NodeStatus.INVALID
         else:
-            (
-                _,
-                _,
-                missing_parents_map,
-                type_inference_failed_columns,
-            ) = validate_node_data(downstream.current, session)
-            if missing_parents_map or type_inference_failed_columns:
-                downstream.current.status = NodeStatus.INVALID
+            # We should not failed node restoration just because of some nodes
+            # that have been invalid already and stay that way.
+            missing_parents_map = None
+            type_inference_failed_columns = None
+            try:
```

Comment:
Nice catch! It seems like a good setup is for `validate_node_data` to return a list of validation errors, and then for any downstream function that calls it to decide whether to raise or skip enforcement of those errors.

Reply:
Yes, I think that's the right pattern. Let me try and rewrite this...

Reply:
Should we just raise inside the `validate_node_data` the same way, regardless of if it's a draft or published node? That can be caught outside of the function as @shangyian described.

Reply:
One minute... let me show you what I have for this.